internal/controller/ols_app_server_reconciliator.go

```diff
@@ -240,3 +245,34 @@ func (r *OLSConfigReconciler) reconcileService(ctx context.Context, cr *olsv1alp
 	r.logger.Info("OLS service reconciled", "service", service.Name)
 	return nil
 }
+
+func (r *OLSConfigReconciler) reconcileTLSSecret(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
+	var secretValues map[string]string
+	foundSecret := &corev1.Secret{}
+	err := r.Client.Get(ctx, client.ObjectKey{Name: OLSCertsSecretName, Namespace: r.Options.Namespace}, foundSecret)
+	if err != nil {
+		return fmt.Errorf("failed to fetch secret: %w", err)
+	}
+	secretValues, err = getSecretContent(r.Client, OLSCertsSecretName, r.Options.Namespace, []string{"tls.key", "tls.crt"})
+	if err != nil {
+		return fmt.Errorf("failed fetching tls certs from secret: %s. error: %w", OLSCertsSecretName, err)
```

Comment:
@vishnuchalla Here too . If we can give a updated message - `failed` seem to lead that its a operator failure. Thinking - `Secret does not have expected tls.key or tls.crt` ?

Reply:
Sounds good. Updated